src/main/InstanceService.ts

```diff
@@ -78,6 +78,67 @@ class InstanceService {
     this.cachedInstances = new LRUCache(this.maximumInstances);
   }
 
+  private attachListeners(account: Account, instance: Instance): void {
+    account.on(PayloadBundleType.TEXT, (payload: PayloadBundleIncoming) => {
+      const linkPreviewContent = payload.content as TextContent;
+      if (linkPreviewContent.linkPreviews) {
+        linkPreviewContent.linkPreviews.forEach(preview => {
+          if (preview.image) {
+            delete preview.image.data;
+          }
+        });
+      }
+      instance.messages.set(payload.id, payload);
+    });
+
+    account.on(PayloadBundleType.ASSET, (payload: PayloadBundleIncoming) => {
+      const metaPayload = instance.messages.get(payload.id);
+      if (metaPayload && payload) {
+        (payload.content as AssetContent).uploaded = (metaPayload.content as AssetContent).uploaded;
+      }
+      instance.messages.set(payload.id, payload);
+    });
+
+    account.on(PayloadBundleType.ASSET_META, (payload: PayloadBundleIncoming) =>
+      instance.messages.set(payload.id, payload)
+    );
+
+    account.on(PayloadBundleType.ASSET_IMAGE, (payload: PayloadBundleIncoming) =>
+      instance.messages.set(payload.id, payload)
+    );
+
+    account.on(PayloadBundleType.PING, (payload: PayloadBundleIncoming) => instance.messages.set(payload.id, payload));
+
+    account.on(PayloadBundleType.MESSAGE_EDIT, (payload: PayloadBundleIncoming) => {
+      const editedContent = payload.content as EditedTextContent;
+      instance.messages.set(payload.id, payload);
+      instance.messages.delete(editedContent.originalMessageId);
+    });
+
+    account.on(PayloadBundleType.CLEARED, () => {
+      const messages = instance.messages.getAll();
+      for (const message in messages) {
+        instance.messages.delete(message);
+      }
```

Comment:
After https://github.com/wireapp/wire-web-packages/pull/1124 is merged we can also use
```ts
instance.messages.deleteAll()
```